python/pyspark/worker.py

```diff
@@ -531,9 +528,10 @@ def func(_, it):
                         f"the 'terminate' method: {str(e)}"
                     )
 
-    ser = BatchedSerializer(CPickleSerializer(), 100)
+    # Each row is a group so do not batch but send one by one.
```

Comment:
Thanks for catching this. Just curious, if a UDTF only yields a small number of rows (< 5) for each input row, would setting the batch size to be 1 impact the performance?

Reply:
Yeah it would does because the batch is too small. That's why we need a better implementation that creates a batch based on the actual row number.